### ISSUE TITLE
Remove Transformation `is_train`

### ISSUE THREAD

**Issue by jaheba** [awslabs/gluon-ts#1514]
Currently, `Transformation.__call__` takes `is_train` as input to allow for different behaviour when the transformation is applied:

```py
class Transformation(metaclass=abc.ABCMeta):
    """
    Base class for all Transformations.

    A Transformation processes works on a stream (iterator) of dictionaries.
    """

    @abc.abstractmethod
    def __call__(self, data_it: Iterable[DataEntry], is_train: bool) -> Iterable[DataEntry]:
        pass
```

There are multiple disadvantages of this approach:

* Transformations which behave the same in and outside of a training context need to handle an additional argument
* It is not obvious from the outside if a Transformation will behave differently
* There is only `is_train`, any other difference in behaviour needs to be handled elsewhere.

The idea was that estimators just have to create one pipeline, which is used for both training and inference.

`GluonEstimator` specifies the following interface:

```py
def create_transformation(self) -> Transformation:
    ...
```

## Proposal

Instead of having transformation behave differently, we create different pipelines for each data-channel:

```py
def create_transformation(self, channel) -> Transformation:
    ...
    if channel == "train":
        transformations.append(...)
    else:
        transformations.append(...)
```

This should make it more obvious which pipeline-steps behave differently in the different modes, while also simplifying the interface of `Transformation`.

**Comment by lostella** [comment]
I agree that we should in general have transformation blocks that do not "hide" different behavior behind flags. Maybe we can focus first on simplifying the `Transformation` interface (i.e. get rid of the flag) and then see where we need to allow for the "bifurcation" in the behaviour construction. Currently, this happens in the methods that return data loaders for different purposes: maybe that's where the difference should be? Do we even need the `create_transformation` method as part of the estimator API? In the end, data loaders are what is really needed, so the method to construct them could take care of breaking the transformation where is needed

**Comment by jaheba** [comment]
One place where `is_train` is used is feature-generation, where `is_train` indicates that we want to have `len(target)` amount of features, and otherwise add additional `prediction_length` number of values.

```python
prediction_length = 12

add_age = AddAgeFeature("target", "age", prediction_length)
add_age(data, is_train=True/False)

# vs

if is_train:
    add_age = AddAgeFeature("target", "age")
else:
    add_age = AddAgeFeature("target", "age", prediction_length=prediction_length)
```

**Comment by jaheba** [comment]
> Do we even need the create_transformation method as part of the estimator API? In the end, data loaders are what is really needed, so the method to construct them could take care of breaking the transformation where is needed

Looking at the code, I would say no. `create_transformation` is called to just pass it then to `create_training_data_loader`, etc.

`ValidationDataLoader` and `InferenceDataLoader` are also the same, except one passes `is_train=True`.

**Comment by jaheba** [comment]
From what I can see, `is_train` is used in:

* `CanonicalInstanceSplitter`
* `SampleTargetDim` (no-op outside of train)
* `CDFtoGaussianTransform`

Features:

* `AddConstFeature`
* `AddTimeFeatures`
* `AddAgeFeature`
* `AddAggregateLags` (class is never used)
